### PR TITLE
CODENVY-268 Rework authorization from RoleBased to PermissionsBased for WorkspaceAPI

### PR DESCRIPTION
**Pull request by sleshchenko**
**_1 Upvote**_ @skabashnyuk @garagatyi @evoevodin @gazarenkov

### CODE REVIEW

**Comment by gazarenkov** [comment]
OK

**Comment by skabashnyuk** [comment]
ok

**Comment by voievodin** [comment]
+1

File: core/che-core-api-core/src/main/java/org/eclipse/che/everrest/CheMethodInvokerFilter.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ * @author Sergii Leschenko
+ */
+public abstract class CheMethodInvokerFilter implements MethodInvokerFilter {
+    @Override
+    public void accept(GenericMethodResource genericMethodResource, Object[] arguments) throws WebApplicationException {
+        try {
+            filter(genericMethodResource, arguments);
+        } catch (ApiException exception) {
+            Response response;
+            if (exception instanceof ForbiddenException)
```
Comment (garagatyi) [inline core/che-core-api-core/src/main/java/org/eclipse/che/everrest/CheMethodInvokerFilter.java]: Please use curly brackets.

**Pull request merged**